### DISCUSSION TITLE
G21?

### DISCUSSION THREAD

**Discussion by BackwardEcho** [aegean-odyssey/mpmd_marlin_1.1.x#95] [General]
Is `G20` and `G21` not supported? I will never use `G20` as my start gcode has `G21`, but it's two sides of the measurement coin. I'm asking because I am getting 
">>> g21
SENDING:G21
echo:Unknown command: "G21" "

from pronterface, so I am curious if it is just always set to metric (`G21`).

**Comment by mulcmu** [comment]
`INCH_MODE_SUPPORT` is not defined in the configuration.h file so neither `G20` or `G21` will be processed and units will always default to metric.

https://github.com/aegean-odyssey/mpmd_marlin_1.1.x/blob/dbdb71bdab85ba03bc4b1475f5901921b3f48ec9/marlin_changes/Configuration.h#L1605-L1610

https://github.com/aegean-odyssey/mpmd_marlin_1.1.x/blob/ce6fa96df1f36c2bfd835b76332a82ebda8028d5/Marlin-1.1.x/Marlin/Marlin_main.cpp#L12863-L12867

https://github.com/aegean-odyssey/mpmd_marlin_1.1.x/blob/ce6fa96df1f36c2bfd835b76332a82ebda8028d5/Marlin-1.1.x/Marlin/Marlin_main.cpp#L4018-L4028

**Comment by BackwardEcho** [comment]
Oh cool, Thanks for letting me know. I may just remove that from my start gcode then since it doesn't work.

**Comment by BackwardEcho** [comment]
Actually, while you are here @mulcmu, could I bother you for a bit? There are 3 things I am hoping for you to do since I believe you are more experienced on these matters.

1) Could you add Matthew Upp's circuit diagram to the wiki page about the extra fan header? The one in #76 that he made / used when adding the mosfet upside down to help him visualize and orient the component. I think it would be valuable to those few who wish to add the components themselves.

I asked in 76 but everyone seemed to have just stopped responding after the 28th. I don't know how to work github's formatting otherwise I would have added it a long time ago. >_>

2) Could to check to see if Linear Advance is still entirely there and enabled? I am trying to calibrate it but I don't see any difference in the line from a K value of 0.75 to 2. When I say no difference, I mean absolutely no difference to the point it's making me wonder if it's malfunctioning. I believe I have Marlin's pattern generator setup properly to use my .25 nozzle, so unless it just doesn't work for that size I have no clue why else it might not be working. I do have the quiet board but the extruder should be the 4988 driver still.

I checked the advanced configuration.h file and saw the "Enable Linear Adv." line uncommented, so I think it is actually enabled, but I have no idea about the rest of the function. If need be I could just open an issue on it.

3) After doing a M303 PID tune, Pronterface comes back with an "Unknown command MM105". I have no idea where the M303 stuff is stored to see if there is just a typo for `M105`, or even if `M105` is even supported. I would appreciate if you could either point me in the right direction or even just find it.

This one is really not that important, just a nitpick.

**Comment by mulcmu** [comment]
I will encourage you to give editing the wiki a shot yourself.  The online editor for the wiki markdown is very similar to the comments/discussion and has a preview to show the changes before saving.  @aegean-odyssey says "All are welcome and encouraged to join in, participate, and have fun" on the main page.

I haven't experimented with linear advance on the MPMD or used anything but a 0.4 nozzle.  I'd maybe try a calibration print from 0.1 to 1 to see if there are any notable differences in that range or maybe increase the speed a bit.

It seems like the MM105 is not just a typo somewhere based on search of code for that string.  Here was where the `M303` function is defined if you want to have a look:

https://github.com/aegean-odyssey/mpmd_marlin_1.1.x/blob/ce6fa96df1f36c2bfd835b76332a82ebda8028d5/Marlin-1.1.x/Marlin/Marlin_main.cpp#L10394-L10402

I vaguely remember seeing this occur at other times as well but it didn't seem too troublesome to investigate further.

**Comment by BackwardEcho** [comment]
I think I tried before but got scared of mucking up what was already there. I can try again, but don't know how well it will go since the preview doesn't show the images.


edit: Well it's added, but is it by no means pretty. Took me so many revisions of the page just to get frustrated and add it the hard way of uploading a smaller image.

**Reply by mulcmu** [reply]
The finished product looks good to me.  Nice work
